src/monattack.cpp

```diff
@@ -5121,6 +5165,153 @@ bool mattack::bio_op_takedown( monster *z )
+        // Handle mons earlier - less to check for
+        target->deal_damage( z, bp_torso, damage_instance( DT_STAB, dam ) );
+        if( do_bleed ) {
+            target->add_effect( effect_bleed, rng( 75_turns, 125_turns ), bp_torso, true );
+        }
+        if( seen ) {
+            add_msg( _( "The %1$s impales %2$s!" ), z->name(), target->disp_name() );
+        }
+        target->check_dead_state();
+        return true;
+    }
+
+    body_part hit = target->get_random_body_part();
+
+    t_dam = foe->deal_damage( z, hit, damage_instance( DT_STAB, dam ) ).total_damage();
+    
+    target->add_msg_player_or_npc( _( "The %1$s tries to impale your %s…" ),
+        _( "The %1$s tries to impale <npcname>'s %s…" ),
+        z->name(), body_part_name_accusative( hit ) );
+
+    if( t_dam > 0 ) {
+        target->add_msg_if_player( m_bad, _( "and deals %d damage!" ), t_dam );
+
+        if( do_bleed ) {
+            target->as_character()->make_bleed( hit, rng( 75_turns, 125_turns ), true );
+        }
+    } else {
+        target->add_msg_player_or_npc( _( "but fails to penetrate your armor!" ),
+            _( "but fails to penetrate <npcname>'s armor!" ) );
+    }
+
+    target->on_hit( z, hit, z->type->melee_skill );
+    foe->check_dead_state();
+
+    return true;
+}
+
+bool mattack::bio_op_disarm( monster *z )
+{
+    if( !z->can_act() ) {
+        return false;
+    }
+
+    Creature *target = z->attack_target();
+    if( target == nullptr ||
+        !is_adjacent( z, target, false ) ||
+        !z->sees( *target ) ) {
+        return false;
+    }
+
+    const bool seen = g->u.sees( *z );
+    player *foe = dynamic_cast< player * >( target );
+
+    // disarm doesn't work on creatures
+    if( foe == nullptr ) {
+        return false;
+    }
+
+    if( seen ) {
+        add_msg( _( "The %1$s mechanically reaches for %2$s!" ), z->name(),
+            target->disp_name() );
+    }
+    z->moves -= 100;
+
+    if( target->uncanny_dodge() ) {
+        return true;
+    }
+
+    // Can we dodge the attack? Uses player dodge function % chance (melee.cpp)
+    if( dodge_check( z, target ) ) {
+        target->add_msg_player_or_npc( _( "You dodge it!" ),
+            _( "<npcname> dodges it!" ) );
+        target->on_dodge( z, z->type->melee_skill * 2 );
+        return true;
+    }
+
+    int mon_stat = z->type->melee_dice * z->type->melee_sides;
+    int my_roll = dice( 3, 2 * mon_stat );
+    my_roll += dice( 3, z->type->melee_skill );
+
+    /** @EFFECT_STR increases chance to avoid disarm, primary stat */
+    /** @EFFECT_DEX increases chance to avoid disarm, secondary stat */
+    /** @EFFECT_PER increases chance to avoid disarm, secondary stat */
+    /** @EFFECT_MELEE increases chance to avoid disarm */
+    int their_roll = dice( 3, 2 * foe->get_str() + foe->get_dex() );
+    their_roll += dice( 3, foe->get_per() );
+    their_roll += dice( 3, foe->get_skill_level( skill_melee ) );
+
+    item &it = foe->weapon;
```

Comment:
Have you tested what happens (and what is printed) when the player is not wielding anything?

Reply:
Not with that method by itself. I have it setup it can only be called by the random technique method and it will skip the call if the player has no weapon.

Reply:
You have added it via ` add_hardcoded_attack`, so anybody can make a monster with that specific special attack, without going through `:bio_op_random_biojutsu`.